lib/Amazon/S3/Thin.pm

```diff
@@ -278,42 +278,169 @@ You can also pass any useragent as you like
+There are already some useful modules like L<Amazon::S3>, L<Net::Amazon::S3>
+and L<AWS::S3> on CPAN. They provide a "Perlish" interface, which is easy to
+understand for Perl programmers, but they also hide low-level behaviors.
+For example, the "get_key" method returns C<undef> on 404 status and raises
+an exception on 5xx status.
 
-In some situations, it is very important to see raw HTTP communications.
+In some situations, it is very important to see the raw HTTP communications.
 That's why I made this module.
 
-=head1 TO DO
+=head1 CONSTRUCTOR
+
+=head2 new( \%params )
+
+B<Receives:> hashref with options.
+
+B<Returns:> Amazon::S3::Thin object
+
+It can receive the following arguments:
+
+=over 4
+
+=item * C<aws_access_key_id> (B<REQUIRED>) - the access key id
+for your S3 account.
+
+=item * C<aws_secret_access_key> (B<REQUIRED>) - the access key secret
+for your S3 account.
+
+=item * C<secure> - whether to use https or not. Default is 0 (http).
+
+=item * C<host> - the base host to use. Default is 'I<s3.amazonaws.com>'.
+
+=item * C<ua> - a user agent object, compatible with LWP::UserAgent.
+Default is an instance of L<LWP::UserAgent>.
+
+=back
+
+=head1 ACCESSORS
+
+The following accessors are provided. You can use them to get/set your
+object's attributes.
+
+=head2 secure
+
+Whether to use https (1) or http (0) when connecting to S3.
+
+=head2 host
+
+The base host to use for connecting to S3.
+
+=head2 ua
+
+The user agent used internally to perform requests and return responses.
+If you set this attribute, please make sure you do so with an object
+compatible with L<LWP::UserAgent> (i.e. providing the same interface).
+
+=head1 METHODS
+
+=head2 get_object( $bucket, $key )
+
+B<Arguments>: a string with the bucket name, and a string with the key name.
+
+B<Returns>: an L<HTTP::Response> object for the request. Use the C<content()>
+method on the returned object to read the contents:
+
+    my $res = $s3->get_object( 'my.bucket', 'my/key.ext' );
+
+    if ($res->is_success) {
+        my $content = $res->content;
+    }
+
+The GET operation retrieves objects from Amazon S3.
+
+For more information, please refer to
+L<< Amazon's documentation for GET|http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html >>.
+
+=head2 delete_object( $bucket, $key )
+
+B<Arguments>: a string with the bucket name, and a string with the key name.
+
+B<Returns>: an L<HTTP::Response> object for the request.
+
+The DELETE operation removes the null version (if there is one) of an object
+and inserts a delete marker, which becomes the current version of the
+object. If there isn't a null version, Amazon S3 does not remove any objects.
+
+Use the response object to see if it succeeded or not.
+
+For more information, please refer to
+L<< Amazon's documentation for DELETE|http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectDELETE.html >>.
+
+=head2 copy_object( $src_bucket, $src_key, $dst_bucket, $dst_key )
```

Comment:
OK, but I might change the interface in the future, because I am inspired by the PHP SDK  on designing this module.

http://docs.aws.amazon.com/aws-sdk-php/latest/class-Aws.S3.S3Client.html#_copyObject